library/x509write_csr.c

```diff
@@ -221,36 +225,37 @@ static int mbedtls_x509write_csr_der_internal( mbedtls_x509write_csr *ctx, unsig
 
     /* copy the signature */
     c2 = buf + size;
-    MBEDTLS_ASN1_CHK_ADD( sig_and_oid_len, mbedtls_x509_write_sig( &c2, buf + len,
-                                        sig_oid, sig_oid_len, sig, sig_len ) );
+    MBEDTLS_ASN1_CHK_ADD( sig_and_oid_len, mbedtls_x509_write_sig( &c2,
+                             buf + len, sig_oid, sig_oid_len, sig, sig_len ) );
 
     /* compact oids and signature memory locations */
     c2 -= len;
     memmove( c2, buf, len );
-    memset( buf, 0, c2 - buf);
 
     len += sig_and_oid_len;
     MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( &c2, buf, len ) );
-    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c2, buf, MBEDTLS_ASN1_CONSTRUCTED |
-                                                 MBEDTLS_ASN1_SEQUENCE ) );
+    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c2, buf,
+                          MBEDTLS_ASN1_CONSTRUCTED | MBEDTLS_ASN1_SEQUENCE ) );
+    memset( buf, 0, c2 - buf);
+
     return( (int) len );
 }
 
-int mbedtls_x509write_csr_der( mbedtls_x509write_csr *ctx, unsigned char *buf, size_t size,
-                       int (*f_rng)(void *, unsigned char *, size_t),
-                       void *p_rng )
+int mbedtls_x509write_csr_der( mbedtls_x509write_csr *ctx, unsigned char *buf,
+                               size_t size,
+                               int (*f_rng)(void *, unsigned char *, size_t),
+                               void *p_rng )
 {
     int len;
     unsigned char *sig;
 
     if( ( sig = mbedtls_calloc( 1, MBEDTLS_MPI_MAX_SIZE ) ) == NULL )
     {
-        mbedtls_free( sig );
-        sig = NULL;
         return( MBEDTLS_ERR_ASN1_ALLOC_FAILED );
     }
 
-    len = mbedtls_x509write_csr_der_internal( ctx, buf, size, sig, f_rng, p_rng );
+    len = mbedtls_x509write_csr_der_internal( ctx, buf, size, sig, f_rng,
+                                         p_rng );
```

Comment:
Please fix the indentation here, and perhaps break the previous line a bit earlier, e.g. prior to the `f_rng` parameter.